packages/store/types/tests/selection.lint.ts

```diff
@@ -99,4 +99,45 @@ describe('[TEST]: Action Types', () => {
 
     TestBed.get(CheckSelectorComponent); // $ExpectType any
   });
+
+  it('should be correct passed selectors', () => {
+    @State<object[]>({ name: 'zoo' })
+    class Zoo {}
+
+    @State<object[]>({ name: 'themePark' })
+    class ThemePark {}
+
+    class MyPandaState {
+      @Selector() static pandas0 = (state: string[]): string[] => state; // $ExpectError
+
+      @Selector() // $ExpectType (state: string[]) => string[]
+      static pandas1(state: string[]): string[] {
+        return state;
+      }
+
+      @Selector([]) // $ExpectType (state: string[]) => string[]
+      static pandas2(state: string[]): string[] {
+        return state;
+      }
+
+      @Selector(['Hello', 'World']) // $ExpectType (state: string[]) => string[]
+      static pandas3(state: string[]): string[] {
+        return state;
+      }
+
+      @Selector([Zoo, ThemePark]) // $ExpectType (zoos: object[], themeParks: object[]) => object[]
+      static pandas4(zoos: object[], themeParks: object[]): object[] {
+        return [...zoos, ...themeParks];
+      }
+
+      @Selector([() => {}, function custom() {}, { a: 1, b: 2 }]) // $ExpectType (state: string[]) => string[]
+      static pandas5(state: string[]): string[] {
+        return state;
+      }
+    }
+
+    TestBed.configureTestingModule({
```

Comment:
This is not needed in types tests. The code doesn't get executed, just parsed and compiled.

Reply:
done